c/dao.c

```diff
@@ -111,8 +111,13 @@ static int extract_withdraw_header_index(size_t input_index, size_t *index) {
       return ERROR_ENCODING;
     }
   }
-  if (arg_res.attr < 2) {
-    return ERROR_WRONG_NUMBER_OF_ARGUMENTS;
+
+  /* last position arg is header index */
+  if (arg_res.attr > 2) {
```

Comment:
Why this PR has to update DAO?

Reply:
The dao contract assumes the second witness arg is `header index`, which implies we can't use dao with multisig lock, @xxuejie think to use the last arg to indicate `header index` is a good idea, and we can compatible with exists format.